src/main/jbake/content/docs/components/metrics/index.adoc

```diff
@@ -4,22 +4,467 @@ Thomas Segismont
+toc::[]
 
+== Introduction
+Hawkular Metrics is a metrics storage engine built on top of link:http://cassandra.apache.org[Cassandra] that offers
+a rich, growing set of features that includes:
 
-Three types of metrics are currently supported:
+* Multi tenancy
+* Multiple metric types
+* Tagging
+* Automatic data removal
+* REST API
+* Scalability
+* Rich query support
 
- * Availability: metric that represent availability: up or down
- * Counter: numeric value that always goes up; example: the total number of visitors of a web page
-   ** Rates: rate of change can be retrieve for counter rates
- * Gauge: numeric value that can go up or down; example: the response time on an application
-   ** Aggregate buckets (min/max/average on some intervals) can be obtained for gauges when individual datapoints are not needed
+== Identifiers
+All identifiers are stored as variable length, UTF-8 encoded strings. This includes:
 
-Tags are available for easy filtering and querying, along with retention times for automatic pruning of older data.
+* Tenant ids
+* Metric names
+* Tag keys (for both metric and data point tags)
 
+[TIP]
+At present there is no restriction on characters that can be used in identifiers. This may change in the future
+though (See link:https://issues.jboss.org/browse/HWKMETRICS-208[HWKMETRICS-208] for details). For this reason it is
+recommended to restrict the characters to letters, numbers, underscore, period, and forward slash.
 
+== Multi Tenancy
+Hawkular Metrics provides _virtual_ multi tenancy. All data is mapped to a tenant, but the data on disk is not
+physically partitioned by tenant. From an API perspective though, everything is partitioned by tenant. All requests,
+both reads and writes, must include a tenant id.
+
+TIP: Tenants do not have to be explicitly created. They will be created when inserting data points.
+
+=== Tenant Ids
+A tenant has an id that uniquely identifies it. The id is a variable length, UTF-8 encoded string. Hawkular Metrics
+does not perform any validation checks to prevent duplicate ids. This is in large part due to Cassandra's design. Among
+other things, Cassandra is a key/value store. Inserting a row into Cassandra is similar to inserting an entry into a
+map. If the key already exists in the map, it will simply be overwritten with the new value. This is exactly how
+Cassandra behaves.
+
+[IMPORTANT]
+If a duplicate id is used, data will be silently overwritten. Users are responsible for ensuring that tenant ids are
+unique.
+
+== Metric Types
+A metric represents a single time series or continuous stream of data points. A data point consists of a timestamp and
+a value. The type of value will vary depending on the metric type. Three types of metrics are currently supported:
+
+* Availability
+* Gauge
+* Counter
+
+TIP: Metrics do not have to be explicitly created. They will be created when inserting data points.
```

Comment:
I completely forgot that we are still using LWTs in the create tenant/metric methods. I thought I had removed the use of LWTs. Since it is still there, the docs need to reflect the behavior.